IPython/core/debugger.py

```diff
@@ -311,6 +311,17 @@ def interaction(self, frame, traceback):
                 if self.shell.readline is not None:
                     self.shell.readline.set_completer_delims(self.shell.readline_delims)
 
+    def parseline(self, line):
+        if line.startswith("!!"):
+            # Force standard behavior.
+            return super(Pdb, self).parseline(line[2:])
+        # "Smart command mode" from pdb++: don't execute commands if a variable
+        # with the same name exists.
+        cmd, arg, newline = super(Pdb, self).parseline(line)
+        if cmd in self.curframe.f_globals or cmd in self.curframe.f_locals:
+            return super(Pdb, self).parseline("!" + line)
```

Comment:
This is normal pdb behavior: https://docs.python.org/3.5/library/pdb.html#pdbcommand-!